cpp/src/gandiva/engine.h

```diff
@@ -54,6 +54,19 @@ class GANDIVA_EXPORT Engine {
     functions_to_compile_.push_back(fname);
   }
 
+  /// Set BaseObjectCache.
+  template<class KeyType>
+  Status SetLLVMObjectCache(GandivaObjectCache<KeyType>& object_cache){
+    //ARROW_LOG(INFO) << "[OBJ-CACHE-LOG]: Entered the SetLLVMObjectCache().";
+    execution_engine_->setObjectCache(&object_cache);
+    if (execution_engine_->hasError()){
+      return Status::ExecutionError("[OBJ-CACHE-LOG]: Can not set Projector Object cache");
+    } else {
```

Comment:
I think you can move the code outside the `else` and remove the `else` block

Reply:
Done